### PR TITLE
Add local release package support for PHPStan

### PR DESCRIPTION
**Pull request by timur27**
#### Changes proposed in this Pull Request
By default, QIT tests are running against a [published version](https://qit.woo.com/docs/cli/running-tests#testing-a-published-extension), but they can also run against the local build. The latter is beneficial when working on PHPStan reports local and checking the results against your local changes. This PR adds the new `npm run test:qit-phpstan-local` command which will run tests with the QIT CLI `--zip` flag.

#### Testing instructions

* run `npm run test:qit-phpstan` and check the plugin version in the test result, it should be the latest published version of WooPayments
* navigate to `woocommerce-payments.php` and change the `Version` header to be any other random version
* run `npm run test:qit-phpstan-local` and confirm that the test result refers the version you provided in the `woocommerce-payments.php` `Version` header

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8846` or branch name `add/phpstan-support-for-local-zip` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/phpstan-support-for-local-zip) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: abd25004d1e9799cc62988391746cb2f35e5e037
- Build time: 2024-05-21 07:14:03 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/qit/phpstan.sh [hunk 1]
```diff
@@ -6,9 +6,33 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
...
+LOCAL_FLAG=false
+ZIP_FILE=""
+
+for arg in "$@"; do
+  if [[ "$arg" == "--local" ]]; then
+    LOCAL_FLAG=true
+  fi
+done
+
+if [[ "$LOCAL_FLAG" == true ]]; then
```
Comment (frosso) [inline tests/qit/phpstan.sh]:
```suggestion
ZIP_FILE=""

if echo "$@" | grep -q -- "--local"; then
```

(⛏️ ) What do you think about this approach for checking for the `--local` flag?

**Review by frosso — APPROVED** [review]
Working great, both with and without `--local`!

File: tests/qit/phpstan.sh (hunk 1 above)
Comment (timur27) [reply tests/qit/phpstan.sh]: Sounds good, simplified it in 1197c5dd38a1a1e32fc8091dfafedf7ce4394137, thanks!

**Pull request merged**